src/lsdb/core/plotting/skymap.py

```diff
@@ -1,32 +1,61 @@
-from typing import Any, Dict
+from __future__ import annotations
+
+from typing import Any, Callable, Dict
 
 import healpy as hp
 import numpy as np
+import pandas as pd
+from dask import delayed
 from hipscat.pixel_math import HealpixPixel
 
+from lsdb.dask.merge_catalog_functions import filter_by_hipscat_index_to_pixel
+
+
+@delayed
+def perform_inner_skymap(
+    partition: pd.DataFrame,
+    func: Callable[[pd.DataFrame, HealpixPixel], Any],
+    pixel: HealpixPixel,
+    target_order: int,
+    **kwargs,
+) -> np.ndarray:
+    """Splits a partition into pixels at a target order and performs a given function on the new pixels"""
+    delta_order = target_order - pixel.order
+    pixels = np.arange(pixel.pixel << (2 * delta_order), (pixel.pixel + 1) << (2 * delta_order))
+    return np.vectorize(
+        lambda p: func(
+            filter_by_hipscat_index_to_pixel(partition, target_order, p),
+            HealpixPixel(target_order, p),
+            **kwargs,
+        )
+    )(pixels)
+
 
-def plot_skymap(pixel_map: Dict[HealpixPixel, Any], **kwargs):
-    """Plot a map of healpix_pixels with values on a mollweide projection.
+def compute_skymap(pixel_map: Dict[HealpixPixel, Any], order: int | None = None, default_value: Any = 0.0):
+    """Returns a histogram map of healpix_pixels to values.
 
     Args:
         pixel_map(Dict[HealpixPixel, Any]): A dictionary of healpix pixels and their values
-        kwargs: Additional keyword arguments to pass to `healpy.mollview`
+        order (int): The order to make the histogram at
```

Comment:
It would be useful to have the default values for `order` and `default_value` in the docstring